### ISSUE TITLE
Prevent users from closing modals + Close modals programmatically

### ISSUE THREAD

**Issue by MotionPhix** [protonemedia/laravel-splade#179]
Can we have a feature that lets us to be able to close modals from vue? I found myself in a situation where I disabled the close button of the X-SPLADE-MODAL and wanted to close the modal manually. I have failed miserably. Also, can we have a feature that stops a user from accidentally closing the modals? Like say someone is editing a model and then accidentally presses outside the modal. I want the modal to stay there and not close accidentally. I am thinking if we could add a **PERSISTENT** or **CLOSABLE** attribute of the modal to denote that it can only be closed if someone clicks either on the cancel or given buttons.

**Comment by pascalbaljet** [comment]
Good suggestions! I'll work on it.

**Comment by pascalbaljet** [comment]
Did you see that you can close a modal from **within** the modal?

```blade
<x-splade-modal>
    <button type="button" @click="modal.close">Cancel</button>
</x-splade-modal>

**Comment by MotionPhix** [comment]
> Can we have a feature that lets us to be able to close modals from vue? I found myself in a situation where I disabled the close button of the X-SPLADE-MODAL and wanted to close the modal manually. I have failed miserably. Also, can we have a feature that stops a user from accidentally closing the modals? Like say someone is editing a model and then accidentally presses outside the modal. I want the modal to stay there and not close accidentally. I am thinking if we could add a **PERSISTENT** or **CLOSABLE** attribute of the modal to denote that it can only be closed if someone clicks either on the cancel or given buttons.

Are we anywhere close to having this? I mean the auto-closing of the modals when someone clicks outside the modal?

**Comment by MotionPhix** [comment]
So i published splade to take a look into the modal wrapper. Currently it has the folowing code:

```
<!--START-SPLADE-MODAL-{{ $key }}-->
<SpladeModal {{ $baseAttributes->mergeVueBinding(':close-button', $closeButton) }} :name="@js($name)">
  <template #default="modal">
    <x-splade-component is="transition" show="modal.isOpen">
      <x-splade-component is="dialog" v-bind:dusk="`modal.${modal.stack}`" close="modal.setIsOpen" class="relative z-20">
        <!-- The backdrop, rendered as a fixed sibling to the panel container -->
        <x-splade-component is="transition" child v-if="modal.stack === 1 && modal.animate" animation="opacity">
          <div v-show="modal.onTopOfStack" class="fixed z-30 inset-0 bg-black/75" />
        </x-splade-component>

        <div v-if="(modal.stack === 1 && !modal.animate) || (modal.stack > 1 && modal.onTopOfStack)" class="fixed z-30 inset-0 bg-black/75" />

        {{ $slot }}
      </x-splade-component>
    </x-splade-component>
  </template>
</SpladeModal>
<!--END-SPLADE-MODAL-{{ $key }}-->
```

I figured out that if you just remove the part where it says `` close="modal.setIsOpen"``, when you click on the backdrop, the modals won't close automatically. You have to either save the modal or click on cancel to close the modal. Which is what I wanted, btu sometimes you might need a different use case, but so far it works for me.. Pointers to a better implmentationwould be highly appreciated.

**Comment by pascalbaljet** [comment]
This is now available in v1.4.1. You may add the `close-explicitly` attribute to disable closing the modal when the user presses the escape key or clicks outside of the modal:

```blade
<x-splade-modal close-explicitly>
```

You can also set this as the default behavior:

```php
Splade::defaultModalCloseExplicitly();
```

Docs: https://splade.dev/docs/x-modal

**Comment by AliasSuperman** [comment]
This doesn't appear to be working.

I immediately show the modal on page load with the opened attribute. And I'm disabling the close button:

<x-splade-modal close-explicitly opened :close-button="false">

I've also have this in AppServiceProvider:

  public function boot(): void
    {
        SpladeTable::defaultSearchDebounce(1000); //750
        Splade::defaultModalCloseExplicitly();
    }
    
    I can still close the modal by clicking outside it.
